Add a section about JS Doc (#441)
springernature/frontend-playbook: javascript/house-style.md

const uniqueIDs = new Set([1, 2, 1, 2]);
```

#### JS Doc

> Although your JavaScript code should be self-documenting, it is highly

Reviewer: "it is highly recommended" -> "we highly recommend that you" 

(House style, "[Prefer the active voice to the passive voice](https://github.com/springernature/frontend-playbook/blob/main/writing/house-style.md#plain-english)")

> that you to document your JavasScript application or library using the

Reviewer 2: ```suggestion
that you document your JavaScript application or library using the
```

> Although your JavaScript code should be self-documenting, we highly recommend
> that you document your JavaScript application or library using the
> Standardised [JSDoc](https://jsdoc.app/) format.

Reviewer 3: I think this might a bit more clear, as we seem to be suggesting that self-documenting code is the preference while also recommending using JSDoc

```suggestion
Although your JavaScript code should be self-documenting, we highly recommend
that where necessary, you document your JavaScript application or library using the
standardised [JSDoc](https://jsdoc.app/) format.
```

> Standardised [JS Doc](https://jsdoc.app/) format.

Reviewer 2: Respecting their chosen name:

```suggestion
Standardised [JSDoc](https://jsdoc.app/) format.
```

This format is leveraged by the Language Server Protocol (LSP), which is
supported in numerous editors and Integrated Development Environments (IDE)
nowadays.  
Thanks to LSP you get short lines of documentation right from the

> invocation context. This helps you code quickly, while avoid obvious errors.

Reviewer: ```suggestion
invocation context. This helps you code quickly, while avoiding obvious errors.
```

Author: Fixed, thank you

> For functions for instance, you get:

Reviewer 2: Consider using e.g. for example instead of for instance since instance has a touch of ambiguity in that context:

```suggestion
For e.g. functions, you get:
```

- What it is intended to do
- The type, default value, and description of its parameters
- Which parameters are optional
- The function returned value type and description

Bonus: There are tools to generate a documentation website for your application

> or library if it uses JS Doc.

Reviewer 2: ```suggestion
or library if it uses JSDoc.
```

A concrete and furnished example:

```js
/**

>  * Get a list of books from given a library.

Reviewer 2: This reads a bit awkwardly.

Author: Thank you so much @danyalaytekin, I've committed all your great suggestions. I will work on making this one to read less awkwardly 🔎.

Reviewer 4: Yeah thanks for the feedback Danyal! 🤗

Reviewer 2: Thanks both 🙂 I need to have a good read through this playbook again as I keep seeing a lot of good stuff going in.  I don't have permissions to resolve this thread btw (in case you were waiting for me to)

Author: Hi @danyalaytekin, I've just pushed the hopefully improvement to the JSDoc example, in case you do not mind having another look at it. Thank you 🙏. I guess that I can resolve the conversion as soon as you feedback positively on it 🤷

Reviewer 2: @morgaan https://github.com/springernature/frontend-playbook/pull/441/commits/684628849dd7a6c61b64c3038b00d4778f8760da looks great to me!

>  * Get a list of books from given a library.

Reviewer 3: ```suggestion
 * Get a list of books from a given library.
```

 *
 * @param {number} library - Id of the library.
 * @param {number} [limit=10] - Optional maximum of books expected to be returned, defaults to 10.
 * @param {object} [filters={}] - Optional filters, defaults to no filters.
 * @param {string} [filters.subject] - A string to match up the subject of the book.
 * @param {number} [filters.publicationyear] - Year the book was published.
 *

>  * @returns {object[]} list of books from that library matching the optionally given filters.

Reviewer 2: For consistency with the param lines:

```suggestion
 * @returns {object[]} List of books from that library matching the optionally given filters.
```

Or perhaps return a list of ids here, or the reader might wonder why favouritism has been shown to `filters` or assume that describing an array of objects isn't possible:

```suggestion
 * @returns {string[]} List of book ids matching the optionally given filters.
```

Then again, that might make the name of the function imprecise.  A dilemma.

 */

> function getbooks(library, limit = 10, filters = {}) {
>   // ...

Reviewer 2: To match house style:

```suggestion
function getBooks(library, limit = 10, filters = {}) {
    // ...
```
